src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -64,34 +64,39 @@ internal DateTime StartTimeCore
         /// <returns>The converted time.</returns>
         internal static DateTime BootTimeToDateTime(TimeSpan timespanAfterBoot)
         {
-            // Use the uptime and the current time to determine the absolute boot time.
-            DateTime bootTime = DateTime.UtcNow - Uptime;
-
             // And use that to determine the absolute time for timespan.
-            DateTime dt = bootTime + timespanAfterBoot;
+            DateTime dt = BootTime + timespanAfterBoot;
 
             // The return value is expected to be in the local time zone.
             // It is converted here (rather than starting with DateTime.Now) to avoid DST issues.
             return dt.ToLocalTime();
         }
 
-        /// <summary>Gets the elapsed time since the system was booted.</summary>
-        private static TimeSpan Uptime
+        /// <summary>Gets the system boot time.</summary>
+        private static DateTime BootTime
         {
             get
             {
-                // '/proc/uptime' accounts time a device spends in sleep mode.
-                const string UptimeFile = Interop.procfs.ProcUptimeFilePath;
-                string text = File.ReadAllText(UptimeFile);
-
-                double uptimeSeconds = 0;
-                int length = text.IndexOf(' ');
-                if (length != -1)
+                // '/proc/stat -> btime' gets the boot time.
+                // btime is the time of system boot in seconds since the Unix epoch.
+                // It includes suspended time and is updated based on the system time (settimeofday).
+                const string StatFile = Interop.procfs.ProcStatFilePath;
+                string text = File.ReadAllText(StatFile);
+                var btimeLineStart = text.IndexOf("\nbtime ");
+                if (btimeLineStart >= 0)
                 {
-                    Double.TryParse(text.AsSpan(0, length), NumberStyles.AllowDecimalPoint, NumberFormatInfo.InvariantInfo, out uptimeSeconds);
+                    var btimeStart = btimeLineStart + "\nbtime ".Length;
+                    var btimeEnd = text.IndexOf('\n', btimeStart);
+                    if (btimeEnd > btimeStart)
+                    {
+                        if (Int32.TryParse(text.AsSpan(btimeStart, btimeEnd - btimeStart), out var bootTimeSeconds))
+                        {
+                            return new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc) + TimeSpan.FromSeconds(bootTimeSeconds);
+                        }
+                    }
                 }
 
-                return TimeSpan.FromSeconds(uptimeSeconds);
+                return DateTime.UtcNow;
```

Comment:
In what situation would we get here?

Reply:
As I understand this actually shouldn't happen. It will get there if it can't find btime line or can't parse it. And this is equivalent to what is in the original version - it returns Uptime == 0 which is equivalent to BootTime == DateTime.Now

Reply:
Ok, thanks.